cookbook/tools/firecrawl_tools.py

```diff
@@ -1,5 +1,48 @@
 from phi.agent import Agent
 from phi.tools.firecrawl import FirecrawlTools
+from dotenv import load_dotenv
 
-agent = Agent(tools=[FirecrawlTools(scrape=False, crawl=True)], show_tool_calls=True, markdown=True)
-agent.print_response("Summarize this https://finance.yahoo.com/")
+load_dotenv()
+
+
+def agent_with_predefined_toolkit(url: str):
+    agent = Agent(
+        tools=[FirecrawlTools(map=True, scrape=True)],
+        show_tool_calls=True,
+        markdown=True,
+        reasoning=True,
+    )
+    agent.print_response(f"Fetch all the blog links from the website: {url}")
+    agent.print_response(
+        "scrape and summarize the following website: https://www.firecrawl.dev/blog/using-structured-output-and-json-strict-mode-openai"
+    )
+
+
+# TODO: Think about this and validate the feasibility of this support
+# def agent_with_explicit_tools(url: str):
+#     fc_tools = [
+#         FirecrawlTools(map=True, scrape=True).map_website(url, include_subdomains=True, ignore_sitemap=False, limit=10),
+#         FirecrawlTools(map=True, scrape=True).scrape_website(url),
+#     ]
+#     agent = Agent(tools=fc_tools, show_tool_calls=True, markdown=True, reasoning=True)
+#     agent.print_response(f"Fetch all the blog links from the website: {url}")
+#     agent.print_response(
+#         "scrape and summarize the following website: https://www.firecrawl.dev/blog/using-structured-output-and-json-strict-mode-openai"
+#     )
```

Comment:
you want to remove this or keep it?

Reply:
wanted to have a discussion on this first with you, and proceed accordingly.

Reply:
Let me know what discussion you want to have

Reply:
Can we access the tools or methods that are defined within a toolkit directly in any way?
For eg. if I want only to use `map_website` and `scrape_website` and pass them in my function calling, is there a way to do this?
This is important to give users the flexibility and control over the tools they can pass in the tool call/function calling. Let me know if anything that's not clear from my end.
cc: @manthanguptaa

Reply:
check for the if conditions you introduced in the FirecrawlTools class. Where the functions are only registered when the params are passed as True.

Reply:
got it. Thanks